vignettes/vetiver.Rmd

```diff
@@ -0,0 +1,150 @@
+---
+title: "Version, share, and deploy a model with vetiver"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{vetiver}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+Sys.setenv("TESTTHAT" = "true")
+options(pins.quiet = FALSE)
+```
+
+The goal of vetiver is to provide fluent tooling for MLOps tasks for your trained model including:
+
+- versioning
+- storing
+- sharing
+- deploying
+
+## Create a `vetiver_model()`
+
+The vetiver package is extensible, with generics that can support many kinds of models. For this example, let's consider one kind of model supported by vetiver, a [tidymodels](https://www.tidymodels.org/) workflow that encompasses both feature engineering and model estimation.
+
+```{r}
+library(parsnip)
+library(recipes)
+library(workflows)
+library(dplyr)
+data(bivariate, package = "modeldata")
+bivariate_train
+
+biv_rec <-
+  recipe(Class ~ ., data = bivariate_train) %>%
+  step_BoxCox(all_predictors())%>%
+  step_normalize(all_predictors())
+
+svm_spec <-
+  svm_linear(mode = "classification") %>%
+  set_engine("LiblineaR")
+
+svm_fit <- 
+  workflow(biv_rec, svm_spec) %>%
+  fit(sample_frac(bivariate_train, 0.7))
+```
+
+This `svm_fit` object is a fitted workflow, with both feature engineering and model parameters estimated using the training data `bivariate_train`. We can create a `vetiver_model()` from this trained model; a `vetiver_model()` collects the information needed to store, version, and deploy a trained model.
+
+```{r}
+library(vetiver)
+v <- vetiver_model(svm_fit, "biv_svm")
+v
+```
+
+Think of this `vetiver_model()` as a deployable model object.
+
+## Store and version your model
+
+You can store and version your model by choosing a [pins](https://pins.rstudio.com) "board" for it, including a local folder, RStudio Connect, Amazon S3, and more. Most pins boards have versioning turned on by default, but we can turn it on explicitly for our temporary demo board. When we write the `vetiver_model()` to our board, the binary model object is stored on our board together with necessary metadata, like the packages needed to make a prediction and the model's input data prototype for checking new data at prediction time.
+
+```{r}
+library(pins)
+model_board <- board_temp(versioned = TRUE)
+model_board %>% vetiver_pin_write(v)
+```
+
+Let's train our model again with a new version of the dataset and write it once more to our board.
+
+```{r}
+svm_fit <- 
+  workflow(biv_rec, svm_spec) %>%
+  fit(sample_frac(bivariate_train, 0.7))
+v <- vetiver_model(svm_fit, "biv_svm")
+
+model_board %>% vetiver_pin_write(v)
+```
+
+Both versions are stored, and we have access to both.
```

Comment:
Here I asked myself how we have access to the different versions. It became clear to me later on when the plumber file contains `vetiver_pin_read(version = "some-version-string")`. For someone who knows pins (which is probably the most part of this audience here?) it might be clear up here already though.